File: Task2/apz-pzpi-21-11-milennyi-ivan-task2/src/view_models/sheep_vm.rs
use serde::Serialize;
use sqlx::FromRow;
use utoipa::ToSchema;

#[derive(Debug, Serialize, FromRow, ToSchema)]
pub(crate) struct SheepVM{
    id: u64,
    breed: String,
    sex: bool,
    birth_date: u64,
    last_feeding_timestamp: Option<u64>,
    last_shearing_timestamp: Option<u64>,
}

File: Task4/apz-pzpi-21-11-milennyi-ivan-task4/src/viewModels/SheepVM.ts
import { timestampToDate } from "../utils/helpers";

export class SheepVM {
    id: number;
    breed: string;
    sex: boolean;
    birthDate: number;
    lastFeedingDate: string | null;
    lastShearingDate: string | null;

    constructor(id: number, breed: string, sex: boolean, birthDate: number, lastFeedingTimestamp: number | null, lastShearingTimestamp: number | null) {
        this.id = id;
        this.breed = breed;
        this.sex = sex;
        this.birthDate = birthDate;
        this.lastFeedingDate = (lastFeedingTimestamp != null) ? timestampToDate(lastFeedingTimestamp) : null;
        this.lastShearingDate = (lastShearingTimestamp != null) ? timestampToDate(lastShearingTimestamp) : null;
    }
}